src/allmydata/util/deferredutil.py

```diff
@@ -225,16 +226,20 @@ def until(
             break
 
 
-def async_to_deferred(f):
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def async_to_deferred(f: Callable[P, Awaitable[R]]) -> Callable[P, Deferred[R]]:
     """
     Wrap an async function to return a Deferred instead.
 
     Maybe solution to https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3886
     """
 
     @wraps(f)
-    def not_async(*args, **kwargs):
-        return defer.Deferred.fromCoroutine(f(*args, **kwargs))
+    def not_async(*args: P.args, **kwargs: P.kwargs) -> Deferred[R]:
+        return defer.Deferred.fromCoroutine(f(*args, **kwargs))  # type: ignore
```

Comment:
I wonder what was being ignored here.  Can you make this a more specific ignore comment?

Reply:
Done, and filed upstream issue: https://github.com/twisted/twisted/issues/11832